src/routes/+page.server.ts

```diff
@@ -33,18 +52,32 @@ export const load: PageServerLoad = async (event) => {
 		.where(eq(table.member.userId, event.locals.user.id))
 		.orderBy(desc(table.membership.startTime));
 
-	const memberships = result.map((m) => ({
-		...m.membership,
-		status: m.member.status,
-		unique_id: m.member.id,
-	}));
+	// Get all membership IDs that the user already has
+	const userMembershipIds = new Set(result.map((m) => m.membership.id));
 
-	return { user: event.locals.user, form, memberships };
+	// Check for each membership if there's a next year's membership available
+	const membershipsWithRenewInfo = await Promise.all(
```

Comment:
I think this whole thing could probably be done with a single database query, now it's doing quite many roundtrips potentially, since we first load the memberships and then loop over them in runtime code